# Differential Gene Expression Analysis 

```{r DE-Genes-Setup, include=FALSE}
# Data manipulation
library(dplyr)
library(readr)
library(tibble)
library(magrittr)
library(reshape2)

# Analysis with bioconductor
library(limma)
library(edgeR)
library(RUVSeq)
library(biomaRt)
library(UniProt.ws)
library(genefilter)
library(MSstats)
library(qvalue)

# Visualisation
library(ggplot2)
library(corrplot)
library(RColorBrewer)
library(grid)
library(scales)
library(knitr)
library(kableExtra)
library(pheatmap)
```

```{r DE-Genes-Config, echo=FALSE, message=FALSE}
# Data Directory
dataDir <- file.path("~/Box/Projects/K97fsZebrafishAnalysis/", "data")

# ggplot2 theme
theme_set(theme_bw())

# Viewports for grid
vp_left <- viewport(x = 0, y = 0, width = 0.5, height = 1, just = c(0, 0))
vp_right <- viewport(x = 0.5, y = 0, width = 0.5, height = 1, just = c(0,0))

# Function for saving pheatmap as A4 pdf.
save_pheatmap_pdf <- function(x, filename, width = 8, height = 11) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  pdf(filename, width = width, height = height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
  }

# Pre-load Database objects
mart <- readRDS(file.path(dataDir, "db", "mart.rds")) # Zebrafish BioMart object
up <- readRDS(file.path(dataDir, "db", "up.rds")) # Zebrafish UniProt.ws object
```

```{r DE-Genes-Construct-DGEList-Object-and-Normalise-Data, echo=FALSE, cache=TRUE, message=FALSE}
# ----------------------------------------------------------------------------------#
# geneCountsFile points to the gene count matrix, which is the featureCounts output.
# In the gene count matrix, each row is a gene and each column is an RNA-seq library.
# tgtsFile points to a "targets" file where each row is an RNA-seq library and each 
# column is a sample characteristic or other metadata e.g. age, psen1 genotype, etc.
geneCountsFile <- file.path(dataDir, "de_genes", "K97Gfs_Genes.txt")
tgtsFile <- file.path(dataDir, "de_genes", "tgts.csv")

# ----------------------------------------------------------------------------------#
# Read in the tgts file 
tgts <- read_csv(tgtsFile) %>%
    mutate(months = as.factor(months), 
         age = as.factor(age),
         psen1 = as.factor(psen1),
         date = as.factor(gsub(x = date, 
                               pattern="([0-9]{2})([0-9]{2})([0-9]{2})", 
                               replacement = "\\3-\\2-20\\1")),
         batch = as.factor(batch),
         group = as.factor(group),
         sibship = as.factor(sibship),
         biorep = as.factor(gsub(x = sample, 
                                 pattern = "^.*_S([1-3]).*$", 
                                 replacement = "\\1"))) %>%
  arrange(desc(group), biorep) %>%
  filter(sibship == "K97Gfs") %>%
  droplevels() %>%
  mutate(id = c("6mth_NM_1", "6mth_NM_2", "6mth_NM_3", 
                "24mth_NM_1", "24mth_NM_2", "24mth_NM_3", 
                "6mth_K97_1", "6mth_K97_2", "6mth_K97_3", 
                "24mth_K97_1", "24mth_K97_2", "24mth_K97_3")) %>%
  dplyr::select(sample, months, psen1, date, group, biorep, id)

# Relevel the group column so that the factor levels are:
# K97Gfs_wt_young, K97Gfs_wt_old, K97Gfs_mutant_young, K97Gfs_mutant_old
tgts$group %<>% 
  relevel(ref = "K97Gfs_mutant_young") %>% 
  relevel(ref = "K97Gfs_wt_old") %>% 
  relevel(ref = "K97Gfs_wt_young")

# ----------------------------------------------------------------------------------#
# Read in the gene counts, convert it into a DGEList object called `geneCounts`. 
# Calculate the normalisation factors for each RNAseq library using the default
# Trimmed Mean of M-values (TMM) normalisation method. 
geneCounts <- read_tsv(geneCountsFile, col_names = TRUE, trim_ws = TRUE) %>%
  as.data.frame %>%
  column_to_rownames("Geneid") %>%
  set_colnames(gsub(x = colnames(.), 
                    pattern = "/data/biohub/2017_Lardelli_K97Gfs/4_dedupData/bams/[0-9]{1,2}_|_sorted_dedup_unique.bam", 
                    replacement="")) %>%
  extract(, tgts$sample) %>% #Reorder the sample names in columns to match tgts.
  edgeR::DGEList(remove.zeros = T) %>%
  calcNormFactors(method = "TMM")

# ----------------------------------------------------------------------------------#
# Add the tgts information to `geneCounts$samples`.
geneCounts$samples %<>%
  rownames_to_column("sample") %>%
  left_join(tgts, by="sample") %>%
  mutate(group.x = group.y) %>%
  dplyr::rename(group = group.x) %>%
  dplyr::select(-group.y) %>%
  mutate(group = as.factor(group)) %>%
  as.data.frame() %>%
  column_to_rownames("sample")

# ----------------------------------------------------------------------------------#
# Add zebrafish gene info from BioMart to `geneCounts$genes`.

#getFromBiomart <- c("ensembl_gene_id", "external_gene_name", "chromosome_name",
# "strand", "start_position", "end_position", "gene_biotype", "description")
#geneData <- getBM(getFromBiomart, values = rownames(geneCounts), mart = mart)

geneData <- readRDS(file.path(dataDir, "de_genes", "geneData.rds"))

geneData <- geneData %>%  
  left_join(x = data_frame(ensembl_gene_id = rownames(geneCounts)), 
            y = geneData, 
            by = "ensembl_gene_id")

geneCounts$genes <- geneData %>%
  as.data.frame() %>%
  column_to_rownames("ensembl_gene_id")


```

## Data Description

Total RNA was extracted from the whole brains of *psen1*<sup>K97Gfs/+</sup> and *psen1*<sup>+/+</sup> zebrafish at 6-months and 24-months of age, and subjected to RNA-seq library preparation and RNA-sequencing (RNA-seq). For RNA-seq library (sample) information, see Table \@ref(tab:DE-Genes-Sample-Info).

```{r DE-Genes-Sample-Info, echo=FALSE, message=FALSE, warning=FALSE}
# Display sample information in a table. 
geneCounts$samples %>%
  rownames_to_column("Sample") %>%
  set_colnames(gsub(x = colnames(.), pattern = "(^[[:alpha:]])", replacement = "\\U\\1", perl=TRUE)) %>%
  dplyr::select(Id, Months, Psen1, Group, Lib.size, Norm.factors) %>%
  dplyr::rename(ID = Id, psen1.genotype = Psen1) %>%
  kable(format = "html", caption = "RNA-seq Library Characteristics") %>%
  kable_styling("condensed", font_size = 12) %>%
  scroll_box(width = "770px")
```


## Data Pre-Processing

- Previously, the paired-end reads were aligned to the reference zebrafish genome assembly (GRCz10) using *HISAT2* and reads aligning to each Ensembl gene model were counted with *featureCounts*.
- Gene counts, sample information and gene information (retrieved from *BioMart*) were made into a `DGEList` object and TMM normalisation factors were calculated to account for different RNA-seq library sizes. The distribution of gene counts in each RNA-seq library can be seen in Figure \@ref(fig:Boxplot-Normalised-Gene-Counts).  

```{r Boxplot-Normalised-Gene-Counts, echo=FALSE, warning=FALSE, message=FALSE, fig.cap="Distribution of TMM-normalised gene counts in each RNA-seq library.", out.width="770px", height="385px"}
geneCounts %>% 
  cpm(log = TRUE) %>%
  melt() %>% 
  set_colnames(c("ensembl_gene_id", "sample", "logCPM")) %>%
  left_join(geneCounts$samples%>%rownames_to_column("sample"), by = "sample") %>% 
  ggplot(aes(x = factor(as.character(sample), levels = unique(sample)), y = logCPM, fill = as.factor(group))) + 
  geom_boxplot(outlier.colour = "#888888", outlier.size = 0.25, 
               notch = TRUE, outlier.fill=NULL, outlier.shape = 19,
               show.legend = FALSE) +
  labs(x = "RNA-seq Library", y = "logCPM Gene Count", fill="Group") +
  theme(text = element_text(size = 6),
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(), 
        axis.title.x = element_blank(),
        plot.margin = unit(c(10,1,1,1), "mm")) +
  facet_wrap(~group, scales="free_x", nrow=1) +
  scale_fill_brewer(palette="Paired") 
```

The overall similarity between samples based on their gene expression patterns can be summarised using principal component analysis as shown in Figure \@ref(fig:PCA-Normalised-Gene-Counts). We expect samples to group based on their biological characteristics (e.g. age and *psen1* genotype), rather than other variables like the sequencing date (batch). From the principal component analysis, we observe that:

- There is some separation between **age** groups across the first principal component, indicating that age represents the largest source of variation in gene expression patterns between samples. However, the first principal component only explains < 20% of the variation between samples, indicating that there is much more variation in the samples that cannot be explained by any single variable.
- Across the second principal component, the old (24 month) samples separate by their __*psen1* genotype__ (*psen1*<sup>K97Gfs/+</sup> and *psen1*<sup>+/+</sup>). However, the young (6 month) samples do not separate clearly by their *psen1* genotype, indicating that at a young age, *psen1*<sup>K97Gfs/+</sup> and *psen1*<sup>+/+</sup> samples have overall similar gene expression patterns. 
- It is difficult to tell whether batch effects exist, as there is some confounding between age group and batch (sequencing date). We will try to remove batch effects later and compare the effects of removing batch effects on the differential gene expression results.

```{r PCA-Normalised-Gene-Counts, echo=FALSE, warning=FALSE, message=FALSE, fig.cap="Principal component analysis of gene counts in each RNA-seq library after TMM-normalisation.", out.width="770px"}
pcaAnalysis <- prcomp(t(cpm(geneCounts$counts, log=TRUE))) 
pcaplot <- pcaAnalysis$x %>% extract(, c("PC1", "PC2")) %>%
  as.data.frame() %>%
  rownames_to_column("samples") %>%
  left_join((geneCounts$samples %>% rownames_to_column("samples")), by="samples") %>%
  as_tibble() %>%
  ggplot(aes(PC1, PC2, colour = group, shape = date)) +
  geom_point(size = 2, alpha = 0.6) +
  labs(x = "Principal Component 1 (18.8%)",
       y=  "Principal Component 2 (16.9%)", 
       color = "Group",
       shape = "Batch") +
  scale_shape_manual(values = c(15:18)) + 
  theme(aspect.ratio=1) +
  ggtitle(label = "", subtitle = "PCA using normalised gene counts") +
  scale_color_brewer(palette="Paired") 

pcaplot
```

## Filtering Low Expression Genes

Genes which are expressed at low levels are not very informative for differential expression analysis, as differences between samples may not be biologically relevant (e.g. 1 count in one sample vs. 3 counts in another sample doesn't necessarily mean a 3-fold change). 

In this analysis, we define a gene as having low expression if it is expressed less than 1.5 counts per million (cpm) in half of the total number of samples (in this case, 6 RNA-seq libraries out of the total of 12 RNA-seq libraries). This is equivalent to an average raw count of `r signif(mean((geneCounts$samples$lib.size/1e6)*1.5), 3)` across the RNA-seq libraries. 

The effect of filtering out genes expressed at low levels can be visualised in Figure \@ref(fig:Filtering-Low-Expression-Genes). 

```{r Filtering-Low-Expression-Genes, echo=FALSE, warning=FALSE, message=FALSE, fig.cap="Distributions of gene counts before and after filtering out genes expressed < 1.5 cpm in each RNA-seq library.", out.width="100%"}
# Set the threshold for sufficient gene expression.
keepTheseGenes <- (rowSums(cpm(geneCounts) > 1.5) >= 6) 

# Produce density plots of the logCPM distributions of each
# RNAseq library, before and after filtering. 
beforeFiltering <- geneCounts %>% 
  cpm(log = TRUE) %>% 
  melt %>% 
  filter(is.finite(value)) %>% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle("Before filtering") +
  labs(x = "logCPM", y = "Proportion of Genes") +
  ylim(c(0,0.20))

afterFiltering <- geneCounts %>% 
  cpm(log = TRUE) %>% 
  extract(keepTheseGenes,) %>%
  melt %>% 
  filter(is.finite(value)) %>% 
  ggplot(aes(x = value, colour = Var2)) +
  geom_density() + 
  guides(colour = FALSE) +
  ggtitle("After filtering")+
  labs(x = "logCPM", y = "") +
  ylim(c(0,0.20))

grid.newpage()
print(beforeFiltering, vp = vp_left)
print(afterFiltering, vp  = vp_right)
```
```{r Filtering-Step-For-K97Gfs-Dataset, echo=FALSE}
# Perform the filtering step if the density plots look OK.
geneCounts <- geneCounts[keepTheseGenes,,keep.lib.sizes = FALSE]
```

After filtering, there are `r dim(geneCounts)[1]` genes left from the original `r length(keepTheseGenes)` genes in the dataset. 
The TMM normalisation factors gene counts in each RNA-seq library were also re-calculated to account for this. 

We can visualise the boxplot distributions of gene expression again to visualise the effect of filtering and normalisation in Figure \@ref(fig:Boxplots-After-Filtering). 

```{r Boxplots-After-Filtering-Genes, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Boxplots of gene expression distributions for each sample from the K97Gfs dataset after filtering out low expressed genes and recalculating TMM normalisation factors.", out.width="100%", height="385px"}
geneCounts %>% 
  cpm(log=TRUE) %>%
  melt() %>% 
  set_colnames(c("ensembl_gene_id", "sample", "logCPM")) %>%
  left_join(geneCounts$samples%>%rownames_to_column("sample"), by = "sample") %>% 
  ggplot(aes(x = factor(as.character(sample), levels = unique(sample)), y = logCPM, fill = as.factor(group))) + 
  geom_boxplot(outlier.colour = "#888888", outlier.size = 0.25, 
               notch = TRUE, outlier.fill=NULL, outlier.shape=19,
               show.legend = FALSE) +
  labs(x = "RNA-seq Library", y = "logCPM Gene Count", fill="Group") +
  theme(text = element_text(size = 6), 
        axis.title.x = element_blank(),
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank()) +
  facet_wrap(~group, scales="free_x", nrow = 1) +
  scale_fill_brewer(palette="Paired") 
```

In addition, the principal component analysis can be repeated to visualise the effect of filtering on the overall similarity between RNA-seq libraries. A comparison of the principal component analysis plots before and after filtering is shown in Figure \@ref(fig:PCA-After-Filtering). 

We observe that there is better separation of age groups across the first principal component (which now also represents a larger proportion of the variation between samples). However, the separation across the second principal component is mostly similar. 

```{r PCA-After-Filtering-Genes, echo=FALSE, warning=FALSE, message=FALSE, fig.cap="Principal component analysis plots before and after filtering out low expressed genes and recalculating TMM normalisation factors.", out.width="100%", fig.height=3.5}
# Principal component analysis on the gene counts, taking into 
# account normalisation factors with edgeR's cpm function. 
pcaAnalysis2 <- prcomp(t(cpm(geneCounts, log=TRUE))) 

# Extract principal components 1 and 2, and plot them with
# the samples. The group and sequencing date information
# are obtained by joining the data with the sample information 
# in geneCounts$samples.
pcaplot2 <- pcaAnalysis2$x %>% extract(, c("PC1", "PC2")) %>%
  as.data.frame() %>%
  rownames_to_column("samples") %>%
  left_join((geneCounts$samples %>% rownames_to_column("samples")), by="samples") %>%
  as_tibble() %>%
  ggplot(aes(PC1, PC2, colour = group, shape = date)) +
  geom_point(size = 2, alpha = 0.6) +
  labs(x = "Principal Component 1 (26.22%)",
       y=  "Principal Component 2 (19.52%)", 
       color = "Group",
       shape = "Batch") +
  scale_shape_manual(values = c(15:18)) + 
  theme(legend.position = "none") +
  ggtitle(label = "", subtitle = "PCA using filtered + normalised gene counts" ) +
    scale_color_brewer(palette="Paired") 

# Reformat the original PCA plot from the unfiltered data a little.
pcaplot <- pcaplot + theme(legend.position="none", aspect.ratio = NULL)

grid.newpage()
print(pcaplot, vp = vp_left)
print(pcaplot2, vp  = vp_right)
```

```{r Extract-Top-PC-Genes, echo=FALSE, eval=FALSE}
# Extract the top genes contributing to PC1 variation. Since samples separate
# across PC1 by age, we would expect the samples to cluster based on age-group
# if we extract out the top 100 genes contributing to PC1. 

PC1Genes <- pcaAnalysis2$rotation %>% 
  as.data.frame %>% 
  rownames_to_column("ensembl_gene_id") %>% 
  dplyr::select(ensembl_gene_id, PC1) %>% 
  arrange(desc(PC1)) %>% 
  left_join(geneCounts$genes%>%rownames_to_column("ensembl_gene_id")) %>% 
  filter(!is.na(external_gene_name)) 

geneCounts %>% 
  cpm(log = TRUE) %>% 
  extract(PC1Genes$ensembl_gene_id[1:100],) %>% 
  pheatmap(
    scale="row", cutree_cols = 2,
    clustering_distance_rows = "euclidean", 
    border_color = "white", 
    treeheight_row = 0,
    treeheight_col=10,
    fontsize = 4, 
    color = colorRampPalette(c("#10c1e5", "#82e0b4","#F9F9F9", "#FBB829", "#FF0066"))(100),
    labels_row = PC1Genes$external_gene_name[1:100], 
    labels_col = c(paste0("6_NM_", 1:3), paste0("6_K97_", 1:3), paste0("24_NM_", 1:3), paste0("24_K97_", 1:3)),
  )


# Extract top genes contributing to PC2 variation. 
# This is to investigate whether the two 6 month samples that group away
# from the other samples are likely to be outliers due to technical artifacts.

PC2Genes <- pcaAnalysis2$rotation %>% 
  as.data.frame %>% 
  rownames_to_column("ensembl_gene_id") %>% 
  dplyr::select(ensembl_gene_id, PC2) %>% 
  arrange(desc(PC2)) %>% 
  left_join(geneCounts$genes%>%rownames_to_column("ensembl_gene_id")) %>% 
  filter(!is.na(external_gene_name)) 

geneCounts %>% 
  cpm(log = TRUE) %>% 
  extract(PC2Genes$ensembl_gene_id[1:100], c(1:3,7:9)) %>% 
  pheatmap(
    scale="row", cutree_cols = 2,
    clustering_distance_rows = "euclidean", 
    border_color = "white", 
    treeheight_row = 0,
    treeheight_col=10,
    fontsize = 4, 
    cellheight = 4,
    cellwidth=10,
    color = colorRampPalette(c("#10c1e5", "#82e0b4","#F9F9F9", "#FBB829", "#FF0066"))(100),
    labels_row = PC2Genes$external_gene_name[1:100], 
    labels_col = c(paste0("6_NM_", 1:3), paste0("6_K97_", 1:3)),
  )

# From the heatmap, the two samples do seem to have quite different expression of these genes
# commpared to the other samples. However, not all genes are expressed differently, only some
# seem to be driving the differences on the PCA plot. Because of this, we will still retain the
# samples.
```

## Batch Effect Estimation

Because it is possible that batch effects are present in the data, we will use the *RUVseq* package to estimate one factor of unwanted variation (`k = 1`). The `RUVs` method is suitable since we have 3 biological replicates for each group. 

```{r RUVseq-Batch-Effect-Estimation-Genes, message=FALSE, message=FALSE, warning=FALSE}
# Make a matrix that defines which samples are biological replicates
groupsForRUV <- makeGroups(geneCounts$samples$group) %>% 
  set_rownames(unique(geneCounts$samples$group))

# Apply the RUVs method. 
RUVsOnGeneSet <- RUVs(cpm(geneCounts, log = TRUE), 
                      cIdx = rownames(geneCounts),
                      k = 1, #Estimate 1 factor of unwanted variation. 
                      scIdx = groupsForRUV, isLog=TRUE)

# Extract the covariate W_1 from the RUVs result, and join it to the 
# sample information. 
geneCounts$samples %<>% 
  rownames_to_column("sample") %>% 
  left_join(RUVsOnGeneSet$W %>% as.data.frame %>% rownames_to_column("sample") %>% dplyr::select(sample, W_1)) %>%
  column_to_rownames("sample")
```

*RUVseq* generates a vector `W_1` which contains adjustment factors to correct for the batch effect across samples. This means that the gene counts of samples are not altered directly (which would change the degrees of freedom in the later differential gene expression analysis). Instead, the `W_1` factor for each sample can be added as a covariate to the linear model, and the gene counts are adjusted according to this factor. 

The effect of adding `W_1` as a covariate to the samples can be observed in the principal component analysis plot in Figure \@ref(fig:PCA-after-RUVseq). 

```{r PCA-after-RUVseq, echo=FALSE, fig.cap="Principal component analysis of gene counts from each RNA-seq library after accounting for a batch effect estimated using RUVseq."}
# Use the removeBatchEffect function from limma, including the W_1 factor estimated 
# by RUVseq as a covariate. 
removedBatchMatrix <- removeBatchEffect(cpm(geneCounts, log=TRUE), covariates = geneCounts$samples$W_1)

# Run principal component analysis.
pcaAnalysis3 <- prcomp(t(removedBatchMatrix)) 

# Principal component plot of the first two principal components. 
pcaAnalysis3$x %>% extract(, c("PC1", "PC2")) %>%
  as.data.frame() %>%
  rownames_to_column("samples") %>%
  left_join((geneCounts$samples %>% rownames_to_column("samples")), by="samples") %>%
  as_tibble() %>%
  ggplot(aes(PC1, PC2, colour = group, shape = date)) +
  geom_point(size = 4, alpha = 0.6) +
  labs(x = "Principal Component 1 (32.21%)",
       y=  "Principal Component 2 (12.55%)", 
       color = "Group",
       shape = "Date") +
  scale_shape_manual(values = c(15:18)) + 
  theme(aspect.ratio=1) +
  ggtitle("PCA of batch-corrected gene counts") +
  scale_color_brewer(palette="Paired") 
```

After batch effect modelling, the samples no longer cluster by batch. The wildtype and mutant samples separate out slightly, indicating that their gene expression patterns may overall show some global difference. The differences between old (24 month) wildtype and mutant samples is less clear than before though. 

We will have to compare the results from the *limma* analysis before and after batch effect estimation to see whether batch effect removal is beneficial for this dataset. 


## Limma Analysis 

- Differential gene expression analysis was performed using moderated *t*-tests as implemented in the *limma* package. 
- We will perform the *limma* analysis two times, once including the estimated factor of unwanted variation from earlier (`W_1`) as a covariate in the linear model fitted by *limma*, and once without the `W_1` covariate. The results will then be compared using Venn diagrams.

### Including Batch Effect Estimation

```{r Limma-DE-Gene-Analysis-For-K97Gfs-Dataset, message=FALSE, warning=FALSE, cache=TRUE}
# In this limma analysis, the W_1 covariate is included in the design matrix. 

# Create the design matrix defining  which samples belong to which groups. 
design <- model.matrix(~0 + group + W_1, data = geneCounts$samples) %>%
  set_colnames(gsub(pattern = "group", replacement = "", x = colnames(.)))

# Apply the voom method, which transforms discrete count data into continuous log-normal distribution. 
voomData <- voomWithQualityWeights(geneCounts, design = design, plot = FALSE)
sampleWeights <- data.frame(rownames(voomData$targets), voomData$sample.weights) %>% 
  set_colnames(c("samples", "weights"))

# Define the contrasts (comparisons) which we will test for differential expression.
contrasts <- makeContrasts(
  levels = colnames(design), 
  K97_mutYoung_wtYoung = K97Gfs_mutant_young - K97Gfs_wt_young,
  K97_mutOld_wtOld = K97Gfs_mutant_old - K97Gfs_wt_old,
  K97_wtOld_wtYoung = K97Gfs_wt_old - K97Gfs_wt_young,
  K97_mutOld_mutYoung = K97Gfs_mutant_old - K97Gfs_mutant_young
)

# Do moderated t-test for each gene to test for differential expression. 
# The empirical Bayes step "borrows" information across genes to increase accuracy of variance estimation.
fit <- lmFit(voomData, design) %>%
  contrasts.fit(contrasts) %>%
  eBayes()
#plotSA(fit, main = "Final model: Mean-variance trend")

# Apply fdr adjustment and adjusted p-value cutoff of 0.05 to define significance.
results <- decideTests(fit, p.value = 0.05, adjust.method = "fdr", method= "global")

#summary(results) 
# saveRDS(voomData, "data/de_genes/voomData.rds")
# saveRDS(geneCounts, "data/de_genes/geneCounts.rds")
# saveRDS(fit, "data/de_genes/fit.rds")
```

### Not Including Batch Effect Estimation

```{r Limma-Analysis-No-Batch-Effect-Estimation, cache=TRUE, warning=FALSE, message=FALSE}
# In this limma analysis, the W_1 covariate is not included in the design matrix. 
# Everything else is the same though.

design_nb <- model.matrix(~0 + group, data = geneCounts$samples) %>%
  set_colnames(gsub(pattern = "group", replacement = "", x = colnames(.)))

voomData_nb <- voomWithQualityWeights(geneCounts, design = design_nb, plot = FALSE)

contrasts_nb <- makeContrasts(
  levels = colnames(design_nb), 
  K97_mutYoung_wtYoung = K97Gfs_mutant_young - K97Gfs_wt_young,
  K97_mutOld_wtOld = K97Gfs_mutant_old - K97Gfs_wt_old,
  K97_wtOld_wtYoung = K97Gfs_wt_old - K97Gfs_wt_young,
  K97_mutOld_mutYoung = K97Gfs_mutant_old - K97Gfs_mutant_young
)

fit_nb <- lmFit(voomData, design_nb) %>%
  contrasts.fit(contrasts_nb) %>%
  eBayes()

results_nb <- decideTests(fit_nb, p.value = 0.05, adjust.method = "fdr", method = "global")

#summary(results_nb)
```

### Effect of Accounting for Batch Effects

In Figure \@ref(fig:Comparison), the differentially expressed genes identified from the *limma* analyses accounting for batch effects (`B`) and not accounting for batch effects (`NB`) are compared. 

Overall, accounting for batch effects retains the majority of the differentially expressed genes identified when not accounting for batch effects, while additionally resulting in the identification of many more genes that would have been missed without batch effect removal. 

For the **Results** section coming up, we will use the results identified from the *limma* analysis taking into account the batch effects.

```{r Comparison-DE-Genes-Venn-Diagrams, echo=FALSE, fig.cap="Comparison of differentially expressed genes when not accounting for batch effects (NB) and when accounting for batch effects (B).", fig.height=6, out.width="100%"}
withBatchEstimation <- results %>% as.data.frame %>% set_colnames(gsub(x = colnames(.), pattern = "K97_", replacement = "B_")) 
noBatchEstimation <- results_nb %>% as.data.frame %>% set_colnames(gsub(x = colnames(.), pattern = "K97_", replacement = "NB_")) 

par(mar = c(0,0,0,0), mfrow=c(2,2), pty="m")
cbind(withBatchEstimation, noBatchEstimation) %>%
  dplyr::select(contains("mutYoung_wtYoung")) %>%
  vennDiagram(cex= .5, mar = rep(0.0001,4), lwd = 1, circle.col = c("turquoise", "pink"))

cbind(withBatchEstimation, noBatchEstimation) %>% dplyr::select(contains("mutOld_wtOld")) %>%
  vennDiagram(cex= .5, mar = rep(0.0001,4), lwd = 1, circle.col = c("turquoise", "pink"))

cbind(withBatchEstimation, noBatchEstimation) %>% dplyr::select(contains("wtOld_wtYoung")) %>%
  vennDiagram(cex= .5, mar = rep(0.0001,4), lwd = 1, circle.col = c("turquoise", "pink"))

cbind(withBatchEstimation, noBatchEstimation) %>% dplyr::select(contains("mutOld_mutYoung"))%>%
  vennDiagram(cex= .5, mar = rep(0.0001,4), lwd = 1, circle.col = c("turquoise", "pink"))
```

## Results

```{r Write-Limma-Results-DE-Genes, echo=FALSE}
# This function is a slight modification of limma's write.fit function to 
# allow the final data frame to be retained as an R object rather than 
# written out to an external file. 
write_fit <- function (fit, results = NULL, digits = 3, adjust = "none", 
    method = "separate", F.adjust = "none", sep = "\t", ...) 
{
    if (!is(fit, "MArrayLM")) 
        stop("fit should be an MArrayLM object")
    if (!is.null(results) && !is(results, "TestResults")) 
        stop("results should be a TestResults object")
    if (is.null(fit$t) || is.null(fit$p.value)) 
        fit <- eBayes(fit)
    method <- match.arg(method, c("separate", "global"))
    p.value <- as.matrix(fit$p.value)
    if (adjust == "none") {
        p.value.adj <- NULL
    }
    else {
        p.value.adj <- p.value
        if (method == "separate") 
            for (j in 1:ncol(p.value)) p.value.adj[, j] <- p.adjust(p.value[, 
                j], method = adjust)
        if (method == "global") 
            p.value.adj[] <- p.adjust(p.value, method = adjust)
    }
    if (F.adjust == "none" || is.null(fit$F.p.value)) 
        F.p.value.adj <- NULL
    else F.p.value.adj <- p.adjust(fit$F.p.value, method = F.adjust)
    rn <- function(x, digits = digits) if (is.null(x)) 
        NULL
    else {
        if (is.matrix(x) && ncol(x) == 1) 
            x <- x[, 1]
        round(x, digits = digits)
    }
    tab <- list()
    tab$A <- rn(fit$Amean, digits = digits - 1)
    tab$Coef <- rn(fit$coef, digits = digits)
    tab$t <- rn(fit$t, digits = digits - 1)
    tab$p.value <- rn(p.value, digits = digits + 2)
    tab$p.value.adj <- rn(p.value.adj, digits = digits + 3)
    tab$F <- rn(fit$F, digits = digits - 1)
    tab$F.p.value <- rn(fit$F.p.value, digits = digits + 2)
    tab$F.p.value.adj <- tab$F.p.value.adj <- rn(F.p.value.adj, 
        digits = digits + 3)
    tab$Res <- unclass(results)
    tab$Genes <- fit$genes
    tab <- data.frame(tab, check.names = FALSE)
    #write.table(tab, file = file, quote = FALSE, row.names = FALSE, 
        #sep = sep, ...)
    return(tab)
}

# Write the limma results
limma_results <- write_fit(fit, results = results, digits = 9, adjust = "fdr", method = "global") %>%
  rownames_to_column("ensembl_gene_id")
```

Genes were defined as being differentially expressed for each comparison if the FDR-adjusted p-value was less than 0.05. The differentially expressed genes for each comparison can be visualised as volcano plots in Figures \@ref(fig:Volcano-Plots-Effect-of-psen1-mutation) and \@ref(fig:Volcano-Plots-Effect-of-Age). 

```{r Volcano-Plots-Effect-of-psen1-mutation-DE-Genes, echo=FALSE, out.width="100%", fig.cap="Effect of the K97Gfs psen1 mutation on gene expression in zebrafish brains at 6-months and 24-months of age. The grey dashed line represents a raw p-value of 0.05 while the black line corresponds to an FDR-adjusted p-value cutoff of 0.05.", fig.height=3.5}
DEColours <- c("-1" = "darkturquoise", "1" = "orange", "0" = "#BBBBBB")

volcano_mutYoung_wtYoung <- limma_results %>%
  ggplot(aes(x = Coef.K97_mutYoung_wtYoung, 
             y = -log10(p.value.K97_mutYoung_wtYoung), 
             colour = as.factor(as.character(Res.K97_mutYoung_wtYoung)))) +
  geom_point(size= 0.5, alpha = 0.5) +
  scale_colour_manual(values = DEColours) +
  theme(legend.position = "none") +
  geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
  geom_hline(yintercept = -log10(limma_results %>% 
                                   filter(Res.K97_mutYoung_wtYoung != 0) %>% 
                                   use_series(p.value.K97_mutYoung_wtYoung) %>% max), 
             color = "black") +
  labs(x = "log2 Fold Change", y = "-log10 p-value") +
  ggtitle("Mutant Young vs. Wildtype Young") 

volcano_mutOld_wtOld <- limma_results %>%
  ggplot(aes(x = Coef.K97_mutOld_wtOld, 
             y = -log10(p.value.K97_mutOld_wtOld), 
             colour = as.factor(as.character(Res.K97_mutOld_wtOld)))) +
  geom_point(size= 0.5, alpha = 0.5) +
  scale_colour_manual(values = DEColours) +
  theme(legend.position = "none") +
  geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
  geom_hline(yintercept = -log10(limma_results %>% 
                                   filter(Res.K97_mutOld_wtOld != 0) %>% 
                                   use_series(p.value.K97_mutOld_wtOld) %>% max), 
             color = "black") +
  labs(x = "log2 Fold Change", y = "-log10 p-value") +
  ggtitle("Mutant Old vs. Wildtype Old") 

grid.newpage()
print(volcano_mutYoung_wtYoung, vp = vp_left)
print(volcano_mutOld_wtOld, vp = vp_right)
```

```{r Volcano-Plots-Effect-of-Age, echo=FALSE, out.width="100%", fig.cap="Effect of aging in wildtypes and in K97Gfs psen1 mutants. The grey dashed line represents a raw p-value of 0.05 while the black line corresponds to an FDR-adjusted p-value cutoff of 0.05.", fig.height=3.5}
volcano_wtOld_wtYoung <- limma_results %>%
  ggplot(aes(x = Coef.K97_wtOld_wtYoung, 
             y = -log10(p.value.K97_wtOld_wtYoung), 
             colour = as.factor(as.character(Res.K97_wtOld_wtYoung)))) +
  geom_point(size= 0.5, alpha = 0.5) +
  scale_colour_manual(values = DEColours) +
  theme(legend.position = "none") +
  geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
  geom_hline(yintercept = -log10(limma_results %>% 
                                   filter(Res.K97_wtOld_wtYoung != 0) %>% 
                                   use_series(p.value.K97_wtOld_wtYoung) %>% max), 
             color = "black") +
  labs(x = "log2 Fold Change", y = "-log10 p-value") +
  ggtitle("Wildtype Old vs. Wildtype Young") 

volcano_mutOld_mutYoung <- limma_results %>%
  ggplot(aes(x = Coef.K97_mutOld_mutYoung, 
             y = -log10(p.value.K97_mutOld_mutYoung), 
             colour = as.factor(as.character(Res.K97_mutOld_mutYoung)))) +
  geom_point(size= 0.5, alpha = 0.5) +
  scale_colour_manual(values = DEColours) +
  theme(legend.position = "none") +
  geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
  geom_hline(yintercept = -log10(limma_results %>% 
                                   filter(Res.K97_mutOld_mutYoung != 0) %>% 
                                   use_series(p.value.K97_mutOld_mutYoung) %>% max), 
             color = "black") +
  labs(x = "log2 Fold Change", y = "-log10 p-value") +
  ggtitle("Mutant Old vs. Wildtype Old") 

grid.newpage()
print(volcano_wtOld_wtYoung, vp = vp_left)
print(volcano_mutOld_mutYoung, vp = vp_right)
```

We can use the following heatmaps in Figures \@ref(fig:Heatmap-mutYoung-wtYoung), \@ref(fig:Heatmap-mutOld-wtOld), \@ref(fig:Heatmap-wtOld-wtYoung) and \@ref(fig:Heatmap-mutOld-mutYoung) to look at the expression of the differentially expressed genes in more detail.

```{r Heatmap-mutYoung-wtYoung, fig.height=8, echo=FALSE, fig.cap="Differentially expressed genes between K97Gfs mutants and wildtypes at 6 months."}
DEGenes <- list()
DEGenes[["mutYoung_wtYoung"]] <- limma_results %>% 
  filter(Res.K97_mutYoung_wtYoung != 0, abs(Coef.K97_mutYoung_wtYoung) > 0.5)
DEGenes[["mutOld_wtOld"]] <- limma_results %>%
  filter(Res.K97_mutOld_wtOld != 0, abs(Coef.K97_mutOld_wtOld) > 0.5)
DEGenes[["wtOld_wtYoung"]] <- limma_results %>%
  filter(Res.K97_wtOld_wtYoung != 0, abs(Coef.K97_wtOld_wtYoung) > 0.5) %>%
  arrange(p.value.K97_wtOld_wtYoung)
DEGenes[["mutOld_mutYoung"]] <- limma_results %>%
  filter(Res.K97_mutOld_mutYoung != 0, abs(Coef.K97_mutOld_mutYoung) > 0.5) %>%
  arrange(p.value.K97_mutOld_mutYoung)

heatmap_mutYoung_wtYoung <- geneCounts %>%
  cpm(log = TRUE) %>%
  extract(DEGenes$mutYoung_wtYoung$ensembl_gene_id, c(1:3, 7:9)) %>%
  pheatmap(
    scale="row", 
    gaps_col = c(3),
    cluster_cols = FALSE,
    clustering_distance_rows = "euclidean", 
    border_color = "white", 
    treeheight_row = 0,
    cutree_rows = 2,
    fontsize = 6, 
    cellheight = 7,
    cellwidth = 7,
    color = colorRampPalette(c("#10c1e5", "#82e0b4","#F9F9F9", "#FBB829", "#FF0066"))(100),
    labels_row = DEGenes$mutYoung_wtYoung$Genes.external_gene_name, 
    labels_col = c(paste0("6_NM_", 1:3), paste0("6_K97_", 1:3)),
  )
  
# save_pheatmap_pdf(heatmap_mutYoung_wtYoung,
#                   "../ZebrafishAnalysis/data/2017-10-10/DEGenes_mutYoung_wtYoung_Heatmap.pdf")

DEGenes$mutYoung_wtYoung %>%
  dplyr::select(Genes.external_gene_name, Genes.gene_biotype, Genes.description, 
                Coef.K97_mutYoung_wtYoung, p.value.adj.K97_mutYoung_wtYoung) %>%
  set_colnames(c("Gene", "Biotype", "Description", "logFC","FDR")) %>%
  kable(format = "html") %>%
  kable_styling("condensed") %>%
  scroll_box(width = "770px", height = "770px")
```

<hr>

```{r Heatmap-mutOld-wtOld, echo=FALSE, fig.height=11, fig.cap="Differentially expressed genes between K97Gfs mutants and wildtypes at 24 months. Due to the large number of genes, only genes with absolute logFC > 0.5 were included in this heatmap."}
heatmap_mutOld_wtOld <- geneCounts %>%
  cpm(log = TRUE) %>%
  extract(DEGenes$mutOld_wtOld$ensembl_gene_id, c(4:6, 10:12)) %>%
  pheatmap(
    scale="row", 
    gaps_col = c(3),
    cluster_cols = FALSE,
    clustering_distance_rows = "euclidean", 
    border_color = "white", 
    treeheight_row = 0,
    cutree_rows = 2,
    fontsize = 4, 
    cellheight = 5.5,
    cellwidth = 5.5,
    color = colorRampPalette(c("#10c1e5", "#82e0b4","#F9F9F9", "#FBB829", "#FF0066"))(100),
    labels_row = DEGenes$mutOld_wtOld$Genes.external_gene_name, 
    labels_col = c(paste0("24_NM_", 1:3), paste0("24_K97_", 1:3)),
  )

#save_pheatmap_pdf(heatmap_mutOld_wtOld, "../ZebrafishAnalysis/data/2017-10-10/DEGenes_mutOld_wtOld_Heatmap.pdf")


DEGenes$mutOld_wtOld %>%
  dplyr::select(Genes.external_gene_name, Genes.gene_biotype, Genes.description, 
                Coef.K97_mutOld_wtOld, p.value.adj.K97_mutOld_wtOld) %>%
  set_colnames(c("Gene", "Biotype", "Description", "logFC","FDR")) %>%
  kable(format = "html") %>%
  kable_styling("condensed") %>%
  scroll_box(width = "770px", height = "770px")
```

<hr>

```{r Heatmap-wtOld-wtYoung, echo=FALSE, fig.height=11, fig.cap="Differentially expressed genes between 24 month and 6 month old wildtype zebrafish brains. Due to the large number of genes, the top 100 genes (ranked by p-value) with absolute logFC > 0.5 were included in this heatmap."}
heatmap_wtOld_wtYoung <- geneCounts %>%
  cpm(log = TRUE) %>%
  extract(DEGenes$wtOld_wtYoung$ensembl_gene_id[1:100], c(1:6)) %>%
  pheatmap(
    scale="row", 
    gaps_col = c(3),
    cluster_cols = FALSE,
    clustering_distance_rows = "euclidean", 
    border_color = "white", 
    treeheight_row = 0,
    cutree_rows = 2,
    fontsize = 6, 
    cellheight = 7,
    cellwidth = 7,
    color = colorRampPalette(c("#10c1e5", "#82e0b4","#F9F9F9", "#FBB829", "#FF0066"))(100),
    labels_row = DEGenes$wtOld_wtYoung$Genes.external_gene_name[1:100], 
    labels_col = c(paste0("6_NM_", 1:3), paste0("24_NM_", 1:3))
  )

#save_pheatmap_pdf(heatmap_wtOld_wtYoung,  "../ZebrafishAnalysis/data/2017-10-10/DEGenes_wtOld_wtYoung_Heatmap.pdf")


DEGenes$wtOld_wtYoung %>%
  dplyr::select(Genes.external_gene_name, Genes.gene_biotype, Genes.description, 
                Coef.K97_wtOld_wtYoung, p.value.adj.K97_wtOld_wtYoung) %>%
  set_colnames(c("Gene", "Biotype", "Description", "logFC","FDR")) %>%
  kable(format = "html") %>%
  kable_styling("condensed") %>%
  scroll_box(width = "770px", height = "770px")

```

<hr>

```{r Heatmap-mutOld-mutYoung, echo=FALSE, fig.height=11, fig.cap="Differentially expressed genes between 24 month and 6 month old K97Gfs mutant zebrafish brains. Due to the large number of genes, the top 100 genes (ranked by p-value) with absolute logFC > 0.5 were included in this heatmap."}
heatmap_mutOld_mutYoung <- geneCounts %>%
  cpm(log = TRUE) %>%
  extract(DEGenes$mutOld_mutYoung$ensembl_gene_id[1:100], c(1:6)) %>%
  pheatmap(
    scale="row", 
    gaps_col = c(3),
    cluster_cols = FALSE,
    clustering_distance_rows = "euclidean", 
    border_color = "white", 
    treeheight_row = 0,
    cutree_rows = 2,
    fontsize = 6, 
    cellheight = 7,
    cellwidth = 7,
    color = colorRampPalette(c("#10c1e5", "#82e0b4","#F9F9F9", "#FBB829", "#FF0066"))(100),
    labels_row = DEGenes$mutOld_mutYoung$Genes.external_gene_name[1:100], 
    labels_col = c(paste0("6_K97_", 1:3), paste0("24_K97_", 1:3))
  )

#save_pheatmap_pdf(heatmap_mutOld_mutYoung,  "../ZebrafishAnalysis/data/2017-10-10/DEGenes_mutOld_mutYoung_Heatmap.pdf")

DEGenes$mutOld_mutYoung %>%
  dplyr::select(Genes.external_gene_name, Genes.gene_biotype, Genes.description, 
                Coef.K97_mutOld_mutYoung, p.value.adj.K97_mutOld_mutYoung) %>%
  set_colnames(c("Gene", "Biotype", "Description", "logFC","FDR")) %>%
  kable(format = "html") %>%
  kable_styling("condensed") %>%
  scroll_box(width = "770px", height = "770px")
```

<hr>

## Heatmap Summary

The following code was used to produce the heatmaps in **Figure 1** showing the logFC of differentially expressed genes between *psen1*<sup>K97fs/+</sup> and *psen1*<sup>+/+</sup> zebrafish at 6 months and at 24 months of age, along with the logFC of the other comparisons. 

Note that the code to produce the heatmap is not evaluated for this Markdown document as the figure is too wide to fit on the screen. 


```{r Heatmaps-Figure, eval=FALSE}
# View the genes which are differentially expressed between young wt and mutants.
heatmap_6mth <- limma_results %>% 
  filter(p.value.adj.K97_mutYoung_wtYoung < 0.05, abs(Coef.K97_mutYoung_wtYoung) > 0.5) %>% 
  dplyr::select(Genes.external_gene_name, starts_with("Coef")) %>%
  as.data.frame %>%
  arrange(desc(Coef.K97_mutYoung_wtYoung), desc(Coef.K97_wtOld_wtYoung)) %>%
  column_to_rownames("Genes.external_gene_name") %>%
  dplyr::select("Coef.K97_mutYoung_wtYoung", "Coef.K97_wtOld_wtYoung", "Coef.K97_mutOld_wtOld", "Coef.K97_mutOld_mutYoung")

# To prevent very large or small logFC from standing out too much in the heatmap,
# set the maximum and minimum logFC for visualisation purposes.
heatmap_6mth[(heatmap_6mth)>3] <- 3
heatmap_6mth[(heatmap_6mth)<(-3)] <- (-3)


# Create the heatmap for DE genes at 6 months
h6 <- heatmap_6mth%>%
  t %>%
  pheatmap(
    scale="none",
    cluster_rows = FALSE,
    clustering_method = "complete",
    clustering_distance_cols = "binary",
    border_color = "white", 
    treeheight_col = 60, 
    treeheight_row = 0,
    fontsize = 6, 
    cellheight = 10, cellwidth = 10,
        color = colorRampPalette(c("#10c1e5", "#82e0b4","#F9F9F9", "#FBB829", "#FF0066"))(100))


# Do the same to create heatmap for the 24 month old mutant vs. wild type DE genes
heatmap_24mth <- limma_results %>%
  filter(p.value.adj.K97_mutOld_wtOld < 0.05, abs(Coef.K97_mutOld_wtOld) > 0.5) %>%
  dplyr::select(Genes.external_gene_name, starts_with("Coef")) %>%
  as.data.frame %>%
  arrange(desc(Coef.K97_mutOld_wtOld)) %>%
  column_to_rownames("Genes.external_gene_name") %>%
  dplyr::select("Coef.K97_mutOld_wtOld", "Coef.K97_wtOld_wtYoung", "Coef.K97_mutYoung_wtYoung", "Coef.K97_mutOld_mutYoung")

heatmap_24mth[(heatmap_24mth)>3] <- 3
heatmap_24mth[(heatmap_24mth)<(-3)] <- (-3)

h24 <- heatmap_24mth %>%
  t %>%
  pheatmap(
    scale = "none",
    clustering_distance_cols = "maximum",
    clustering_method = "complete",
    cluster_rows = FALSE,
    border_color = "white",
    treeheight_col = 60, 
    treeheight_row = 0,
    fontsize = 6,
    cellheight = 10, cellwidth = 10,
    labels_col = (limma_results%>%filter(p.value.adj.K97_mutOld_wtOld < 0.05, abs(Coef.K97_mutOld_wtOld) > 0.5))$Genes.external_gene_name,
    color = colorRampPalette(c("#10c1e5", "#82e0b4","#F9F9F9", "#FBB829", "#FF0066"))(100)
  )


```



```{r Extract-Inverted-Genes, eval=FALSE, echo=FALSE}
# Extract genes showing an "inversion" pattern between mutants and wildtypes at
# 6 months and 24 months of age for further motif analysis later. 
# invertedGenes <- limma_results %>% 
#   filter(p.value.adj.K97_mutYoung_wtYoung < 0.15, 
#          p.value.adj.K97_mutOld_wtOld < 0.15) %>%
#   filter((Coef.K97_mutYoung_wtYoung > 0 & Coef.K97_mutOld_wtOld < 0) | 
#            (Coef.K97_mutYoung_wtYoung < 0 & Coef.K97_mutOld_wtOld > 0))

invertedFromYoung <- limma_results %>% 
  filter((Res.K97_mutYoung_wtYoung == 1 & p.value.K97_mutOld_wtOld < 0.01 & Coef.K97_mutOld_wtOld < 0) | (Res.K97_mutYoung_wtYoung == -1 & p.value.K97_mutOld_wtOld < 0.01 & Coef.K97_mutOld_wtOld > 0)) 

invertedFromOld <- limma_results %>% 
  filter((Res.K97_mutOld_wtOld == 1 & p.value.K97_mutYoung_wtYoung < 0.01 & Coef.K97_mutYoung_wtYoung < 0) | (Res.K97_mutOld_wtOld == -1 & p.value.K97_mutYoung_wtYoung < 0.01 & Coef.K97_mutYoung_wtYoung > 0)) 

full_join(invertedFromYoung, invertedFromOld, by = "ensembl_gene_id") %>% 
  distinct(ensembl_gene_id, .keep_all = TRUE) %>% write_tsv("data/2017-10-20/robustInverted.txt")
```


## Session Info

```{r}
sessionInfo()
```

