<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>K97fs Zebrafish Analysis</title>
  <meta name="description" content="K97fs Zebrafish Analysis">
  <meta name="generator" content="bookdown 0.5 and GitBook 2.6.7">

  <meta property="og:title" content="K97fs Zebrafish Analysis" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="K97fs Zebrafish Analysis" />
  
  
  

<meta name="author" content="Nhi Hin">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="directory-setup.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Placeholder</a></li>
<li class="chapter" data-level="2" data-path="differential-gene-expression-analysis.html"><a href="differential-gene-expression-analysis.html"><i class="fa fa-check"></i><b>2</b> Differential Gene Expression Analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="differential-gene-expression-analysis.html"><a href="differential-gene-expression-analysis.html#data-description"><i class="fa fa-check"></i><b>2.1</b> Data Description</a></li>
<li class="chapter" data-level="2.2" data-path="differential-gene-expression-analysis.html"><a href="differential-gene-expression-analysis.html#data-pre-processing"><i class="fa fa-check"></i><b>2.2</b> Data Pre-Processing</a></li>
<li class="chapter" data-level="2.3" data-path="differential-gene-expression-analysis.html"><a href="differential-gene-expression-analysis.html#filtering-low-expression-genes"><i class="fa fa-check"></i><b>2.3</b> Filtering Low Expression Genes</a></li>
<li class="chapter" data-level="2.4" data-path="differential-gene-expression-analysis.html"><a href="differential-gene-expression-analysis.html#batch-effect-estimation"><i class="fa fa-check"></i><b>2.4</b> Batch Effect Estimation</a></li>
<li class="chapter" data-level="2.5" data-path="differential-gene-expression-analysis.html"><a href="differential-gene-expression-analysis.html#limma-analysis"><i class="fa fa-check"></i><b>2.5</b> Limma Analysis</a><ul>
<li class="chapter" data-level="2.5.1" data-path="differential-gene-expression-analysis.html"><a href="differential-gene-expression-analysis.html#including-batch-effect-estimation"><i class="fa fa-check"></i><b>2.5.1</b> Including Batch Effect Estimation</a></li>
<li class="chapter" data-level="2.5.2" data-path="differential-gene-expression-analysis.html"><a href="differential-gene-expression-analysis.html#not-including-batch-effect-estimation"><i class="fa fa-check"></i><b>2.5.2</b> Not Including Batch Effect Estimation</a></li>
<li class="chapter" data-level="2.5.3" data-path="differential-gene-expression-analysis.html"><a href="differential-gene-expression-analysis.html#effect-of-accounting-for-batch-effects"><i class="fa fa-check"></i><b>2.5.3</b> Effect of Accounting for Batch Effects</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="differential-gene-expression-analysis.html"><a href="differential-gene-expression-analysis.html#results"><i class="fa fa-check"></i><b>2.6</b> Results</a></li>
<li class="chapter" data-level="2.7" data-path="differential-gene-expression-analysis.html"><a href="differential-gene-expression-analysis.html#heatmap-summary"><i class="fa fa-check"></i><b>2.7</b> Heatmap Summary</a></li>
<li class="chapter" data-level="2.8" data-path="differential-gene-expression-analysis.html"><a href="differential-gene-expression-analysis.html#session-info"><i class="fa fa-check"></i><b>2.8</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DAProteins.html"><a href="DAProteins.html"><i class="fa fa-check"></i><b>3</b> Differential protein abundance analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="DAProteins.html"><a href="DAProteins.html#manipulating-data"><i class="fa fa-check"></i><b>3.1</b> Manipulating data</a></li>
<li class="chapter" data-level="3.2" data-path="DAProteins.html"><a href="DAProteins.html#protein-analysis"><i class="fa fa-check"></i><b>3.2</b> Protein analysis</a></li>
<li class="chapter" data-level="3.3" data-path="DAProteins.html"><a href="DAProteins.html#visualisation-and-plots"><i class="fa fa-check"></i><b>3.3</b> Visualisation and plots</a></li>
<li class="chapter" data-level="3.4" data-path="DAProteins.html"><a href="DAProteins.html#data-description-1"><i class="fa fa-check"></i><b>3.4</b> Data Description</a></li>
<li class="chapter" data-level="3.5" data-path="DAProteins.html"><a href="DAProteins.html#data-pre-processing-1"><i class="fa fa-check"></i><b>3.5</b> Data Pre-Processing</a><ul>
<li class="chapter" data-level="3.5.1" data-path="DAProteins.html"><a href="DAProteins.html#protein-abundance-estimation"><i class="fa fa-check"></i><b>3.5.1</b> Protein Abundance Estimation</a></li>
<li class="chapter" data-level="3.5.2" data-path="DAProteins.html"><a href="DAProteins.html#normalisation-of-6-month-and-24-month-datasets"><i class="fa fa-check"></i><b>3.5.2</b> Normalisation of 6 month and 24 month datasets</a></li>
<li class="chapter" data-level="3.5.3" data-path="DAProteins.html"><a href="DAProteins.html#protein-annotation"><i class="fa fa-check"></i><b>3.5.3</b> Protein Annotation</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="DAProteins.html"><a href="DAProteins.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>3.6</b> Exploratory Data Analysis</a><ul>
<li class="chapter" data-level="3.6.1" data-path="DAProteins.html"><a href="DAProteins.html#correlation-between-6-and-24-month-samples"><i class="fa fa-check"></i><b>3.6.1</b> Correlation between 6 and 24 month samples</a></li>
<li class="chapter" data-level="3.6.2" data-path="DAProteins.html"><a href="DAProteins.html#principal-component-analysis"><i class="fa fa-check"></i><b>3.6.2</b> Principal Component Analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="DAProteins.html"><a href="DAProteins.html#differential-protein-abundance-analysis"><i class="fa fa-check"></i><b>3.7</b> Differential protein abundance analysis</a></li>
<li class="chapter" data-level="3.8" data-path="DAProteins.html"><a href="DAProteins.html#results-1"><i class="fa fa-check"></i><b>3.8</b> Results</a><ul>
<li class="chapter" data-level="3.8.1" data-path="DAProteins.html"><a href="DAProteins.html#effect-of-the-k97fs-mutation-on-protein-abundance"><i class="fa fa-check"></i><b>3.8.1</b> Effect of the K97fs Mutation on Protein Abundance</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="DAProteins.html"><a href="DAProteins.html#heatmap-summary-1"><i class="fa fa-check"></i><b>3.9</b> Heatmap Summary</a></li>
<li class="chapter" data-level="3.10" data-path="DAProteins.html"><a href="DAProteins.html#enrichment-analysis"><i class="fa fa-check"></i><b>3.10</b> Enrichment Analysis</a></li>
<li class="chapter" data-level="3.11" data-path="DAProteins.html"><a href="DAProteins.html#session-info-1"><i class="fa fa-check"></i><b>3.11</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="load-packages.html"><a href="load-packages.html"><i class="fa fa-check"></i><b>4</b> Load Packages:</a><ul>
<li class="chapter" data-level="4.1" data-path="load-packages.html"><a href="load-packages.html#import-gene-sets-from-msigdb"><i class="fa fa-check"></i><b>4.1</b> Import Gene Sets from MSigDB</a></li>
<li class="chapter" data-level="4.2" data-path="load-packages.html"><a href="load-packages.html#map-human-entrezgenes-to-zebrafish-entrezgenes"><i class="fa fa-check"></i><b>4.2</b> Map Human entrezgenes to Zebrafish entrezgenes</a></li>
<li class="chapter" data-level="4.3" data-path="load-packages.html"><a href="load-packages.html#convert-voom-object-de-gene-analysis-ensembl-gene-ids-to-entrezgenes"><i class="fa fa-check"></i><b>4.3</b> Convert voom object (DE gene analysis) Ensembl gene IDs to entrezgenes</a></li>
<li class="chapter" data-level="4.4" data-path="load-packages.html"><a href="load-packages.html#build-gene-set-indexes"><i class="fa fa-check"></i><b>4.4</b> Build gene set indexes</a></li>
<li class="chapter" data-level="4.5" data-path="load-packages.html"><a href="load-packages.html#fast-rotation-gene-set-analysis-fry"><i class="fa fa-check"></i><b>4.5</b> Fast Rotation Gene Set Analysis (<code>fry</code>)</a></li>
<li class="chapter" data-level="4.6" data-path="load-packages.html"><a href="load-packages.html#use-roast-function-to-obtain-estimate-proportions-of-up-and-down-regulated-genes"><i class="fa fa-check"></i><b>4.6</b> Use <code>roast</code> function to obtain estimate proportions of up and down-regulated genes</a></li>
<li class="chapter" data-level="4.7" data-path="load-packages.html"><a href="load-packages.html#heatmap-summary-2"><i class="fa fa-check"></i><b>4.7</b> Heatmap Summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="directory-setup.html"><a href="directory-setup.html"><i class="fa fa-check"></i><b>5</b> Directory Setup</a><ul>
<li class="chapter" data-level="5.1" data-path="directory-setup.html"><a href="directory-setup.html#define-groups"><i class="fa fa-check"></i><b>5.1</b> Define Groups</a></li>
<li class="chapter" data-level="5.2" data-path="directory-setup.html"><a href="directory-setup.html#promoter-motif-analysis"><i class="fa fa-check"></i><b>5.2</b> Promoter Motif Analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="weighted-co-expression-network-analysis.html"><a href="weighted-co-expression-network-analysis.html"><i class="fa fa-check"></i><b>6</b> Weighted Co-expression Network Analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="weighted-co-expression-network-analysis.html"><a href="weighted-co-expression-network-analysis.html#data-import-and-pre-processing"><i class="fa fa-check"></i><b>6.1</b> Data Import and Pre-Processing</a><ul>
<li class="chapter" data-level="6.1.1" data-path="weighted-co-expression-network-analysis.html"><a href="weighted-co-expression-network-analysis.html#zebrafish-data"><i class="fa fa-check"></i><b>6.1.1</b> Zebrafish Data</a></li>
<li class="chapter" data-level="6.1.2" data-path="weighted-co-expression-network-analysis.html"><a href="weighted-co-expression-network-analysis.html#human-data"><i class="fa fa-check"></i><b>6.1.2</b> Human Data</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="weighted-co-expression-network-analysis.html"><a href="weighted-co-expression-network-analysis.html#network-construction"><i class="fa fa-check"></i><b>6.2</b> Network Construction</a><ul>
<li class="chapter" data-level="6.2.1" data-path="weighted-co-expression-network-analysis.html"><a href="weighted-co-expression-network-analysis.html#choice-of-soft-thresholding-power"><i class="fa fa-check"></i><b>6.2.1</b> Choice of Soft-Thresholding Power</a></li>
<li class="chapter" data-level="6.2.2" data-path="weighted-co-expression-network-analysis.html"><a href="weighted-co-expression-network-analysis.html#constructing-the-co-expression-networks"><i class="fa fa-check"></i><b>6.2.2</b> Constructing the co-expression networks</a></li>
<li class="chapter" data-level="6.2.3" data-path="weighted-co-expression-network-analysis.html"><a href="weighted-co-expression-network-analysis.html#module-identification"><i class="fa fa-check"></i><b>6.2.3</b> Module Identification</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">K97fs Zebrafish Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="weighted-co-expression-network-analysis" class="section level1">
<h1><span class="header-section-number">6</span> Weighted Co-expression Network Analysis</h1>
<p>The aims of this analysis include:</p>
<ol style="list-style-type: decimal">
<li>Pre-process gene expression data for zebrafish brain RNA-seq dataset, and separately, for a human early-onset Alzheimer’s disease brain microarray dataset.</li>
<li>Construct separate gene co-expression networks for zebrafish and human data.</li>
<li>Identify modules of co-expressed genes in each network.</li>
<li>Determine the biological significance of modules using functional enrichment analysis and by correlating module expression with sample traits/characteristics.</li>
<li>Determine whether properties of the zebrafish brain co-expression network are preserved in the human brain co-expression network.</li>
<li>Export network for visualisation.</li>
</ol>
<div id="data-import-and-pre-processing" class="section level2">
<h2><span class="header-section-number">6.1</span> Data Import and Pre-Processing</h2>
<div id="zebrafish-data" class="section level3">
<h3><span class="header-section-number">6.1.1</span> Zebrafish Data</h3>
<p>Import the <code>DGEList</code> object previous made in the DE gene analysis, <code>geneCounts</code>. This object contains three data.frames, <code>sample</code> (sample metadata), <code>counts</code> (gene counts in cpm), and <code>genes</code> (gene annotation from <em>Biomart</em>). The data has already been normalised (TMM method), which can be seen in Figure <a href="weighted-co-expression-network-analysis.html#fig:Normalised-Gene-Expression-Boxplots-WGCNA">6.1</a>.</p>
<div class="figure"><span id="fig:Normalised-Gene-Expression-Boxplots-WGCNA"></span>
<img src="_main_files/figure-html/Normalised-Gene-Expression-Boxplots-WGCNA-1.png" alt="Boxplots showing the distribution of gene counts for each RNA-seq library in the K97Gfs dataset." width="672" />
<p class="caption">
Figure 6.1: Boxplots showing the distribution of gene counts for each RNA-seq library in the K97Gfs dataset.
</p>
</div>
<p>According to <em>WGCNA</em>’s <a href="https://labs.genetics.ucla.edu/horvath/CoexpressionNetwork/Rpackages/WGCNA/faq.html">FAQ page</a>, genes can be filtered by mean expression of variance since low-expressed or non-varying genes usually represent noise, which is undesirable when forming a co-expression network. Below, we will filter by both mean expression and variance to retain genes which tend to be more highly expressed and variable across samples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Density plot of TMM-normalized gene expression distribution for each library before filtering. </span>
A &lt;-<span class="st"> </span>geneCounts <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">cpm</span>(<span class="dt">log =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>melt <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(<span class="kw">is.finite</span>(value)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> value, <span class="dt">colour =</span> Var2)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Before filtering&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;logCPM&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;Density&quot;</span>)

<span class="co"># Retain genes which are expressed at sufficiently high levels, at least 1.5 cpm in 6 (half of the total number of) samples.</span>
keepTheseGenes &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">cpm</span>(geneCounts) <span class="op">&gt;</span><span class="st"> </span><span class="fl">1.5</span>) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">6</span>  

<span class="co"># Density plot of the gene expression distribution after filtering out low expressed genes. </span>
B &lt;-<span class="st"> </span>geneCounts <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">cpm</span>(<span class="dt">log =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">extract</span>(keepTheseGenes,) <span class="op">%&gt;%</span>
<span class="st">  </span>melt <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(<span class="kw">is.finite</span>(value)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> value, <span class="dt">colour =</span> Var2)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;After filtering low exp genes&quot;</span>)

<span class="co"># Retain genes with sufficiently high variation in expression across all samples.</span>
<span class="co"># Here we define sufficiently high variation as having a standard deviation above the 25th quantile (so not in bottom 25%).</span>
variableGenes &lt;-<span class="st"> </span><span class="kw">rowSds</span>(<span class="kw">cpm</span>(geneCounts, <span class="dt">log=</span><span class="ot">TRUE</span>)) <span class="op">&gt;</span><span class="st">  </span><span class="kw">quantile</span>(<span class="kw">rowSds</span>(<span class="kw">cpm</span>(geneCounts, <span class="dt">log=</span><span class="ot">TRUE</span>)), <span class="fl">0.25</span>)

<span class="co"># Density plot of gene expression distribution after filtering out low variance genes. </span>
C &lt;-<span class="st"> </span>geneCounts <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">cpm</span>(<span class="dt">log =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">extract</span>(variableGenes,) <span class="op">%&gt;%</span>
<span class="st">  </span>melt <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(<span class="kw">is.finite</span>(value)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> value, <span class="dt">colour =</span> Var2)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;After filtering variable genes&quot;</span>)

<span class="co"># Density plot of gene expression distribution after filtering out both low expressed and low variance genes. </span>
D &lt;-<span class="st"> </span>geneCounts <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">cpm</span>(<span class="dt">log =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">extract</span>(keepTheseGenes<span class="op">&amp;</span>variableGenes, ) <span class="op">%&gt;%</span>
<span class="st">  </span>melt <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(<span class="kw">is.finite</span>(value)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> value, <span class="dt">colour =</span> Var2)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;After filtering both&quot;</span>)

<span class="co"># Plot the effect of filtering low expressed genes, low variation genes, and both.</span>
<span class="kw">grid.newpage</span>()
<span class="kw">print</span>(A, <span class="dt">vp =</span> vp3)
<span class="kw">print</span>(B, <span class="dt">vp =</span> vp4)
<span class="kw">print</span>(C, <span class="dt">vp =</span> vp1)
<span class="kw">print</span>(D, <span class="dt">vp  =</span> vp2)</code></pre></div>
<div class="figure"><span id="fig:Effect-of-Filtering-Gene-Expression"></span>
<img src="_main_files/figure-html/Effect-of-Filtering-Gene-Expression-1.png" alt="Effect of filtering low expressed and low variance genes from the RNA-seq libraries from the K97Gfs dataset." width="672" />
<p class="caption">
Figure 6.2: Effect of filtering low expressed and low variance genes from the RNA-seq libraries from the K97Gfs dataset.
</p>
</div>
<p>After filtering out lowly expressed genes and low variance genes, each RNA-seq library contains 13381 genes. The distributions of gene counts in each RNA-seq library after normalization and filtering are shown in the following boxplots.</p>
<div class="figure"><span id="fig:Plot-Filtered-Gene-Expression"></span>
<img src="_main_files/figure-html/Plot-Filtered-Gene-Expression-1.png" alt="Effect of TMM (Trimmed Mean of M-values) normalisation followed by filtering out low expression and low variance Genes on the gene expression distributions for each RNA-seq library in the K97Gfs dataset." width="672" />
<p class="caption">
Figure 6.3: Effect of TMM (Trimmed Mean of M-values) normalisation followed by filtering out low expression and low variance Genes on the gene expression distributions for each RNA-seq library in the K97Gfs dataset.
</p>
</div>
<p>The <em>WGCNA</em> <a href="https://labs.genetics.ucla.edu/horvath/CoexpressionNetwork/Rpackages/WGCNA/faq.html">FAQ</a> recommends performing a log<sub>2</sub> transformation before transposing to get the matrix into the correct format for analysis. Here, we use <em>edgeR</em>’s <code>cpm()</code> function with a <code>log2</code> transformation, as the <code>cpm()</code> function takes into account the TMM-normalisation factors previously calculated for each library. The final matrix will be stored in an object called <code>zebrafishWGCNAGenes</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">zebrafishWGCNAGenes &lt;-<span class="st"> </span>geneCounts <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">cpm</span>(<span class="dt">log=</span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span><span class="st"> </span>t <span class="op">%&gt;%</span><span class="st"> </span>as.data.frame</code></pre></div>
</div>
<div id="human-data" class="section level3">
<h3><span class="header-section-number">6.1.2</span> Human Data</h3>
<p>The data is Affymetrix Human Gene 1.1 ST Array gene (and transcript) expression data from human post-mortem individuals with and without <em>PSEN1</em>-based familial Alzheimer’s disease. In this dataset, there are 14 control individuals without Alzheimer’s disease or any neurodegeneration or comorbidities, and 7 individuals with familial Alzheimer’s disease due to a mutation in <em>PSEN1</em> (1 with <code>E120G</code> mutation, 2 with <code>V89L</code> mutation, and 4 with <code>V89L</code> mutation).</p>
<p>Processed and raw data can be downloaded from <a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-39420/">ArrayExpress</a> under the accession number <code>E-GEOD-39420</code>. The raw data has been processed through background adjustment, normalisation, and summarisation by Robust Multi-array Analysis (RMA) using the <em>Affy</em> package (version 1.30.0) from <em>Bioconductor</em> on R language (version 2.13.2). This method is considered reasonable, so we will proceed with this analysis using the processed gene expression data.</p>
<p>First we import the relevant files into R.</p>
<ul>
<li><code>humanSamples</code> from the file <code>E-GEOD-39420.sdrf.txt</code> contains metadata (characteristics and comments) about each sample, including their age at death, <em>APOE</em> genotype, Braak stage (a morphological-based measure of neurodegeneration in their brain), sex, and importantly their <em>PSEN1</em> genotype.</li>
<li><code>humanExprs</code> contains the gene expression data for each sample (one sample per column and one probe per row), along with information related to each probe (row), including gene and transcript assignments.</li>
</ul>
<p>In microarray data, there are multiple probes mapping to each gene We will use the <code>collapseRows()</code> function from the <em>WGCNA</em> package to retain only one gene per probe, using the <code>MaxMean</code> method to choose the gene which has the highest mean expression across the samples.</p>
<p>The boxplots below in Figure <a href="weighted-co-expression-network-analysis.html#fig:Gene-Expression-Boxplots-For-Human-EOAD-Dataset">6.4</a> show the gene expression distribution of each human sample.</p>
<div class="figure"><span id="fig:Gene-Expression-Boxplots-For-Human-EOAD-Dataset"></span>
<img src="_main_files/figure-html/Gene-Expression-Boxplots-For-Human-EOAD-Dataset-1.png" alt="Boxplots showing gene expression distribution for all samples from the human early-onset AD dataset." width="672" />
<p class="caption">
Figure 6.4: Boxplots showing gene expression distribution for all samples from the human early-onset AD dataset.
</p>
</div>
<p>We can also visualise the effect of restricting (filtering) the human dataset to include only the genes which were previously determined in the zebrafish dataset to have sufficiently high expression (1.5 cpm in at least 6 out of 12 samples) and variance (with standard deviation greater than the bottom 25% of standard deviations in the dataset). A comparison of the gene expression distribution before and after filtering is shown below.</p>
<p><img src="_main_files/figure-html/Effect-of-Filtering-For-Human-EOAD-Dataset-1.png" width="672" /></p>
</div>
</div>
<div id="network-construction" class="section level2">
<h2><span class="header-section-number">6.2</span> Network Construction</h2>
<div id="choice-of-soft-thresholding-power" class="section level3">
<h3><span class="header-section-number">6.2.1</span> Choice of Soft-Thresholding Power</h3>
<p>The first step in constructing a scale-free gene co-expression requires choosing a soft-thresholding power.</p>
<p>Here, we test a range of possible soft-thresholding powers to find the minimum one which results in approximately scale-free topology for the zebrafish and human gene expression networks. We define an appropriate minimum scale-free topology fitting index R^2 to be at least 0.80.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Choose a set of soft-thresholding powers to test.</span>
powers =<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">seq</span>(<span class="dt">from =</span> <span class="dv">12</span>, <span class="dt">to =</span> <span class="dv">20</span>, <span class="dt">by =</span> <span class="dv">2</span>))

<span class="co"># Call the network topology analysis function.</span>
sft_human =<span class="st"> </span><span class="kw">pickSoftThreshold</span>(human_genes, 
                              <span class="dt">powerVector =</span> powers, 
                              <span class="dt">verbose =</span> <span class="dv">0</span>, 
                              <span class="dt">networkType =</span> <span class="st">&quot;signed&quot;</span>, 
                              <span class="dt">RsquaredCut =</span> <span class="fl">0.80</span>)</code></pre></div>
<pre><code>##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
## 1      1  0.01010  3.000          0.839  4170.0    4190.0   4470
## 2      2  0.01110  0.972          0.965  2420.0    2410.0   2920
## 3      3  0.00481 -0.295          0.833  1530.0    1510.0   2140
## 4      4  0.01440 -0.243          0.667  1040.0    1000.0   1660
## 5      5  0.16300 -0.480          0.693   745.0     690.0   1360
## 6      6  0.53000 -0.812          0.858   554.0     492.0   1190
## 7      7  0.75500 -0.992          0.945   425.0     359.0   1060
## 8      8  0.87000 -1.100          0.974   335.0     267.0    958
## 9      9  0.91900 -1.160          0.983   270.0     202.0    875
## 10    10  0.94300 -1.190          0.984   221.0     155.0    804
## 11    12  0.97200 -1.240          0.990   156.0      94.9    693
## 12    14  0.97800 -1.270          0.985   115.0      60.1    609
## 13    16  0.98200 -1.280          0.985    87.7      39.5    543
## 14    18  0.98300 -1.280          0.983    68.9      26.9    489
## 15    20  0.98300 -1.270          0.981    55.4      18.7    443</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sft_zebrafish =<span class="st"> </span><span class="kw">pickSoftThreshold</span>(zebrafish_genes, 
                                  <span class="dt">powerVector =</span> powers, 
                                  <span class="dt">verbose =</span> <span class="dv">0</span>, 
                                  <span class="dt">networkType =</span> <span class="st">&quot;signed&quot;</span>, 
                                  <span class="dt">RsquaredCut =</span> <span class="fl">0.80</span>)</code></pre></div>
<pre><code>##    Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
## 1      1    0.170  2.79          0.553  4160.0    4180.0   4470
## 2      2    0.210 -3.58          0.550  2400.0    2340.0   2960
## 3      3    0.221 -2.31          0.726  1520.0    1470.0   2170
## 4      4    0.258 -1.79          0.832  1020.0     976.0   1680
## 5      5    0.354 -1.70          0.893   722.0     678.0   1340
## 6      6    0.421 -1.67          0.902   529.0     489.0   1100
## 7      7    0.495 -1.65          0.909   400.0     363.0    924
## 8      8    0.551 -1.65          0.907   310.0     274.0    785
## 9      9    0.597 -1.62          0.908   245.0     211.0    675
## 10    10    0.637 -1.61          0.906   197.0     165.0    587
## 11    12    0.742 -1.54          0.945   133.0     105.0    454
## 12    14    0.822 -1.55          0.977    93.4      70.1    365
## 13    16    0.841 -1.66          0.983    68.1      48.0    310
## 14    18    0.864 -1.72          0.990    51.1      33.9    267
## 15    20    0.881 -1.78          0.994    39.3      24.5    233</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#-------------------------------------------------------------------------------------------#</span>
<span class="co"># Plot the results:</span>

<span class="co">#pdf(&quot;WGCNA/softThreshold.pdf&quot;, width = 11, height = 8)</span>
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>));
cex1 =<span class="st"> </span><span class="fl">0.9</span>;

<span class="co"># Human: Scale-free topology fit index as a function of the soft-thresholding power</span>
<span class="kw">plot</span>(sft_human<span class="op">$</span>fitIndices[,<span class="dv">1</span>], <span class="op">-</span><span class="kw">sign</span>(sft_human<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft_human<span class="op">$</span>fitIndices[,<span class="dv">2</span>],
     <span class="dt">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,<span class="dt">ylab=</span><span class="st">&quot;Scale Free Topology Model Fit,signed R^2&quot;</span>,<span class="dt">type=</span><span class="st">&quot;n&quot;</span>,
     <span class="dt">main =</span> <span class="kw">paste</span>(<span class="st">&quot;Scale independence&quot;</span>))
<span class="kw">text</span>(sft_human<span class="op">$</span>fitIndices[,<span class="dv">1</span>], <span class="op">-</span><span class="kw">sign</span>(sft_human<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft_human<span class="op">$</span>fitIndices[,<span class="dv">2</span>],
     <span class="dt">labels=</span>powers,<span class="dt">cex=</span>cex1,<span class="dt">col=</span><span class="st">&quot;red&quot;</span>)
<span class="co"># this line corresponds to using an R^2 cut-off of h</span>
<span class="kw">abline</span>(<span class="dt">h=</span><span class="fl">0.80</span>,<span class="dt">col=</span><span class="st">&quot;red&quot;</span>)
<span class="co"># Mean connectivity as a function of the soft-thresholding power</span>
<span class="kw">plot</span>(sft_human<span class="op">$</span>fitIndices[,<span class="dv">1</span>], sft_human<span class="op">$</span>fitIndices[,<span class="dv">5</span>],
     <span class="dt">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,<span class="dt">ylab=</span><span class="st">&quot;Mean Connectivity&quot;</span>, <span class="dt">type=</span><span class="st">&quot;n&quot;</span>,
     <span class="dt">main =</span> <span class="kw">paste</span>(<span class="st">&quot;Mean connectivity&quot;</span>))
<span class="kw">text</span>(sft_human<span class="op">$</span>fitIndices[,<span class="dv">1</span>], sft_human<span class="op">$</span>fitIndices[,<span class="dv">5</span>], 
     <span class="dt">labels=</span>powers, <span class="dt">cex=</span>cex1,<span class="dt">col=</span><span class="st">&quot;red&quot;</span>)

<span class="co"># Zebrafish: Scale-free topology fit index as a function of the soft-thresholding power</span>
<span class="kw">plot</span>(sft_zebrafish<span class="op">$</span>fitIndices[,<span class="dv">1</span>], <span class="op">-</span><span class="kw">sign</span>(sft_zebrafish<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft_zebrafish<span class="op">$</span>fitIndices[,<span class="dv">2</span>],
     <span class="dt">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,<span class="dt">ylab=</span><span class="st">&quot;Scale Free Topology Model Fit,signed R^2&quot;</span>,<span class="dt">type=</span><span class="st">&quot;n&quot;</span>,
     <span class="dt">main =</span> <span class="kw">paste</span>(<span class="st">&quot;Scale independence&quot;</span>))
<span class="kw">text</span>(sft_zebrafish<span class="op">$</span>fitIndices[,<span class="dv">1</span>], <span class="op">-</span><span class="kw">sign</span>(sft_zebrafish<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft_zebrafish<span class="op">$</span>fitIndices[,<span class="dv">2</span>],
     <span class="dt">labels=</span>powers,<span class="dt">cex=</span>cex1,<span class="dt">col=</span><span class="st">&quot;blue&quot;</span>);
<span class="co"># this line corresponds to using an R^2 cut-off of h</span>
<span class="kw">abline</span>(<span class="dt">h=</span><span class="fl">0.80</span>,<span class="dt">col=</span><span class="st">&quot;blue&quot;</span>)
<span class="co"># Mean connectivity as a function of the soft-thresholding power</span>
<span class="kw">plot</span>(sft_zebrafish<span class="op">$</span>fitIndices[,<span class="dv">1</span>], sft_zebrafish<span class="op">$</span>fitIndices[,<span class="dv">5</span>],
     <span class="dt">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,<span class="dt">ylab=</span><span class="st">&quot;Mean Connectivity&quot;</span>, <span class="dt">type=</span><span class="st">&quot;n&quot;</span>,
     <span class="dt">main =</span> <span class="kw">paste</span>(<span class="st">&quot;Mean connectivity&quot;</span>))
<span class="kw">text</span>(sft_zebrafish<span class="op">$</span>fitIndices[,<span class="dv">1</span>], sft_zebrafish<span class="op">$</span>fitIndices[,<span class="dv">5</span>], 
     <span class="dt">labels=</span>powers, <span class="dt">cex=</span>cex1,<span class="dt">col=</span><span class="st">&quot;blue&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/Choose-Soft-Thresholding-Power-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#dev.off()</span></code></pre></div>
<p>The soft power chosen will be 14 as this is the minimum one that results in a scale-free network topology <em>R<sup>2</sup></em> value of at least 0.80 for both the zebrafish and human networks.</p>
</div>
<div id="constructing-the-co-expression-networks" class="section level3">
<h3><span class="header-section-number">6.2.2</span> Constructing the co-expression networks</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># 1. Construct adjacency matrices for each dataset, containing each gene&#39;s</span>
<span class="co"># connection strength (similar to correlation) to all other genes. </span>
adjacency_human &lt;-<span class="st"> </span><span class="kw">adjacency</span>(human_genes, <span class="dt">power =</span> <span class="dv">14</span>, <span class="dt">type =</span> <span class="st">&quot;signed&quot;</span>)
adjacency_zebrafish &lt;-<span class="st"> </span><span class="kw">adjacency</span>(zebrafish_genes, <span class="dt">power =</span> <span class="dv">14</span>, <span class="dt">type =</span> <span class="st">&quot;signed&quot;</span>)

<span class="co">#--------------------------------------------------------------------------------------------#</span>
<span class="co"># 2. Calculate whole network connectivity, k, which is defined as the sum of connection</span>
<span class="co"># strengths to other genes for each sample. </span>
connectivity_human &lt;-<span class="st"> </span><span class="kw">apply</span>(adjacency_human, <span class="dv">1</span>, sum)
connectivity_zebrafish &lt;-<span class="st"> </span><span class="kw">apply</span>(adjacency_zebrafish, <span class="dv">1</span>, sum)

<span class="co"># Scale connectivity to be between 0 and 1.</span>
connectivity_human &lt;-<span class="st"> </span>connectivity_human<span class="op">/</span><span class="kw">max</span>(connectivity_human)
connectivity_zebrafish &lt;-<span class="st"> </span>connectivity_zebrafish<span class="op">/</span><span class="kw">max</span>(connectivity_zebrafish) 

<span class="co"># Plot scale free topology. ------------------------------------------------------------------#</span>
<span class="co"># pdf(&quot;data/2017-10-09/scale-free-topology-plots.pdf&quot;, width = 8, height = 4)</span>
<span class="co"># par(mfrow=c(1,2), mar=c(4,4,4,4))</span>
<span class="co"># scaleFreePlot(connectivity_human, main=&quot;Human&quot;)</span>
<span class="co"># scaleFreePlot(connectivity_zebrafish, main=&quot;Zebrafish&quot;)</span>
<span class="co"># dev.off()</span>
<span class="co"># --------------------------------------------------------------------------------------------#</span>

<span class="co"># --------------------------------------------------------------------------------------------#</span>
<span class="co"># 3. Restrict genes in network to high connectivity genes from either dataset to reduce noise</span>
<span class="co"># during network construction.</span>
<span class="co"># Here, high connectivity is defined as being above the 10th percentile.</span>
minConnections &lt;-<span class="st"> </span><span class="fl">0.1</span>
sufficientlyConnected &lt;-<span class="st"> </span>((connectivity_human <span class="op">&gt;</span><span class="st"> </span>minConnections) <span class="op">|</span><span class="st"> </span>(connectivity_zebrafish <span class="op">&gt;</span><span class="st"> </span>minConnections))
<span class="co"># table(sufficientlyConnected) #  genes passing the minimum connectivity threshold. </span>

<span class="co"># Restrict analysis to  genes which pass the minimum connectivity threshold.</span>
adjacency_human &lt;-<span class="st"> </span>adjacency_human[sufficientlyConnected, sufficientlyConnected]
adjacency_zebrafish &lt;-<span class="st"> </span>adjacency_zebrafish[sufficientlyConnected, sufficientlyConnected]
human_genes <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">extract</span>(,<span class="kw">rownames</span>(adjacency_human))
zebrafish_genes <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">extract</span>(, <span class="kw">rownames</span>(adjacency_zebrafish))
geneCounts &lt;-<span class="st"> </span>geneCounts[<span class="kw">rownames</span>(adjacency_zebrafish),,keep.lib.sizes =<span class="st"> </span><span class="ot">FALSE</span>]
<span class="co"># summary(colnames(human_genes)==colnames(zebrafish_genes)) #Check that genes are still in same order for both datasets.</span>
<span class="co"># --------------------------------------------------------------------------------------------#</span>

<span class="co"># 4. Construct topological overlap matrices for the networks. </span>
TOM_human &lt;-<span class="st"> </span><span class="kw">TOMsimilarity</span>(adjacency_human, <span class="dt">TOMType =</span> <span class="st">&quot;signed&quot;</span>, <span class="dt">verbose =</span> <span class="dv">0</span>)
<span class="kw">colnames</span>(TOM_human) &lt;-<span class="st"> </span><span class="kw">rownames</span>(TOM_human) &lt;-<span class="st"> </span><span class="kw">rownames</span>(adjacency_human)

TOM_zebrafish &lt;-<span class="st"> </span><span class="kw">TOMsimilarity</span>(adjacency_zebrafish, <span class="dt">TOMType =</span> <span class="st">&quot;signed&quot;</span>, <span class="dt">verbose =</span> <span class="dv">0</span>)
<span class="kw">colnames</span>(TOM_zebrafish) &lt;-<span class="st"> </span><span class="kw">rownames</span>(TOM_zebrafish) &lt;-<span class="st"> </span><span class="kw">rownames</span>(adjacency_zebrafish)

<span class="co"># 5. Transform the TOM into dissimilarity matrices. </span>
dissTOM_human &lt;-<span class="st"> </span><span class="dv">1</span> <span class="op">-</span><span class="st"> </span>TOM_human
dissTOM_zebrafish &lt;-<span class="st"> </span><span class="dv">1</span> <span class="op">-</span><span class="st"> </span>TOM_zebrafish

<span class="co"># 6. Hierarchial clustering of the dissimilarity TOM:</span>
geneTree_human &lt;-<span class="st"> </span><span class="kw">flashClust</span>(<span class="kw">as.dist</span>(dissTOM_human), <span class="dt">method =</span> <span class="st">&quot;average&quot;</span>)
geneTree_zebrafish &lt;-<span class="st"> </span><span class="kw">flashClust</span>(<span class="kw">as.dist</span>(dissTOM_zebrafish), <span class="dt">method =</span> <span class="st">&quot;average&quot;</span>)</code></pre></div>
</div>
<div id="module-identification" class="section level3">
<h3><span class="header-section-number">6.2.3</span> Module Identification</h3>
<p>To identify modules of co-expressed genes, we use a a</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Initialise the color and module eigengene list vectors for the zebrafish dataset. </span>
colorList_zebrafish &lt;-<span class="st"> </span><span class="kw">list</span>()
MEList_zebrafish &lt;-<span class="st"> </span><span class="kw">list</span>()

<span class="co">#------------------------------------------------------------------------------------#</span>
<span class="co"># METHOD 1: Hybrid Method + PAM</span>

zebrafishHybridCut &lt;-<span class="st"> </span><span class="kw">cutreeHybrid</span>(
  <span class="dt">dendro =</span> geneTree_zebrafish,
  <span class="dt">distM =</span> dissTOM_zebrafish,
  <span class="dt">minClusterSize =</span> <span class="dv">40</span>,
  <span class="dt">deepSplit =</span> <span class="dv">1</span>,
  <span class="dt">pamStage =</span> <span class="ot">TRUE</span>,
  <span class="dt">maxPamDist =</span> <span class="fl">0.90</span>
)</code></pre></div>
<pre><code>##  ..cutHeight not given, setting it to 0.994  ===&gt;  99% of the (truncated) height range in dendro.
##  ..done.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">colorList_zebrafish[[<span class="st">&quot;HybridCut&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">labels2colors</span>(zebrafishHybridCut<span class="op">$</span>labels)

MEList_zebrafish[[<span class="st">&quot;HybridCut&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">moduleEigengenes</span>(zebrafish_genes, colorList_zebrafish<span class="op">$</span>HybridCut) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">use_series</span>(<span class="st">&quot;eigengenes&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">orderMEs</span>()

colorList_zebrafish<span class="op">$</span>HybridCut <span class="op">%&gt;%</span><span class="st"> </span>table <span class="op">%&gt;%</span><span class="st"> </span>as.data.frame <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">set_colnames</span>(<span class="kw">c</span>(<span class="st">&quot;Module&quot;</span>, <span class="st">&quot;Genes&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(<span class="kw">desc</span>(Genes))</code></pre></div>
<pre><code>##           Module Genes
## 1           grey  1553
## 2      turquoise   876
## 3           blue   518
## 4          brown   373
## 5         yellow   318
## 6          green   289
## 7            red   280
## 8          black   258
## 9           pink   257
## 10       magenta   256
## 11        purple   197
## 12   greenyellow   185
## 13           tan   182
## 14        salmon   141
## 15          cyan   138
## 16  midnightblue   138
## 17     lightcyan   135
## 18        grey60   123
## 19    lightgreen   118
## 20   lightyellow   114
## 21     royalblue   112
## 22       darkred   106
## 23     darkgreen   105
## 24      darkgrey    93
## 25 darkturquoise    93
## 26        orange    84
## 27    darkorange    76
## 28         white    75
## 29       skyblue    68
## 30   saddlebrown    59
## 31     steelblue    51</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#------------------------------------------------------------------------------------#</span>
<span class="co"># METHOD 2: Hybrid Method + Merge close modules</span>

zebrafishHybridCut_merged &lt;-<span class="st"> </span><span class="kw">mergeCloseModules</span>(zebrafish_genes, 
                                               colorList_zebrafish<span class="op">$</span>HybridCut,
                                               <span class="dt">MEs =</span> MEList_zebrafish<span class="op">$</span>HybridCut,
                                               <span class="dt">cutHeight =</span> <span class="fl">0.15</span>)</code></pre></div>
<pre><code>##  mergeCloseModules: Merging modules whose distance is less than 0.15
##    Calculating new MEs...</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">colorList_zebrafish[[<span class="st">&quot;HybridCutMerged&quot;</span>]] &lt;-<span class="st"> </span>zebrafishHybridCut_merged<span class="op">$</span>colors
MEList_zebrafish[[<span class="st">&quot;HybridCutMerged&quot;</span>]] &lt;-<span class="st"> </span>zebrafishHybridCut_merged<span class="op">$</span>newMEs

colorList_zebrafish<span class="op">$</span>HybridCutMerged <span class="op">%&gt;%</span><span class="st"> </span>table <span class="op">%&gt;%</span><span class="st"> </span>as.data.frame <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">set_colnames</span>(<span class="kw">c</span>(<span class="st">&quot;Module&quot;</span>, <span class="st">&quot;Genes&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(<span class="kw">desc</span>(Genes))</code></pre></div>
<pre><code>##           Module Genes
## 1           grey  1553
## 2          black  1170
## 3         salmon  1017
## 4          brown   373
## 5      lightcyan   332
## 6         yellow   318
## 7            red   280
## 8           pink   257
## 9        magenta   256
## 10   greenyellow   185
## 11           tan   182
## 12       darkred   181
## 13          cyan   138
## 14  midnightblue   138
## 15        grey60   123
## 16    lightgreen   118
## 17   lightyellow   114
## 18     royalblue   112
## 19      darkgrey    93
## 20 darkturquoise    93
## 21        orange    84
## 22    darkorange    76
## 23       skyblue    68
## 24   saddlebrown    59
## 25     steelblue    51</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">colorList_human &lt;-<span class="st"> </span><span class="kw">list</span>()
MEList_human &lt;-<span class="st"> </span><span class="kw">list</span>()

<span class="co">#------------------------------------------------------------------------------------#</span>
<span class="co"># METHOD 1: HYBRIDCUT, deepSplit = 2</span>

humanHybridCut &lt;-<span class="st"> </span><span class="kw">cutreeHybrid</span>(
  <span class="dt">dendro =</span> geneTree_human,
  <span class="dt">distM =</span> dissTOM_human,
  <span class="dt">minClusterSize =</span> <span class="dv">40</span>,
  <span class="dt">deepSplit =</span> <span class="dv">2</span>,
  <span class="dt">pamStage =</span> <span class="ot">TRUE</span>,
  <span class="dt">maxPamDist =</span> <span class="fl">0.90</span>
)</code></pre></div>
<pre><code>##  ..cutHeight not given, setting it to 0.993  ===&gt;  99% of the (truncated) height range in dendro.
##  ..done.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">colorList_human[[<span class="st">&quot;HybridCut&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">labels2colors</span>(humanHybridCut<span class="op">$</span>labels)
MEList_human[[<span class="st">&quot;HybridCut&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">moduleEigengenes</span>(human_genes, colorList_human<span class="op">$</span>HybridCut) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">use_series</span>(<span class="st">&quot;eigengenes&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">orderMEs</span>()

colorList_human<span class="op">$</span>HybridCut <span class="op">%&gt;%</span><span class="st"> </span>table <span class="op">%&gt;%</span><span class="st"> </span>as.data.frame <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">set_colnames</span>(<span class="kw">c</span>(<span class="st">&quot;Module&quot;</span>, <span class="st">&quot;Genes&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(<span class="kw">desc</span>(Genes))</code></pre></div>
<pre><code>##         Module Genes
## 1    turquoise  2713
## 2         grey  1146
## 3         blue   884
## 4        brown   798
## 5       yellow   313
## 6        green   301
## 7          red   243
## 8        black   241
## 9         pink   165
## 10     magenta   139
## 11      purple   137
## 12 greenyellow   128
## 13         tan   111
## 14      salmon    52</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#------------------------------------------------------------------------------------#</span>
<span class="co"># METHOD 2: HYBRIDCUT, deepSplit = 3</span>

humanHybridCut_<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw">cutreeHybrid</span>(
  <span class="dt">dendro =</span> geneTree_human,
  <span class="dt">distM =</span> dissTOM_human,
  <span class="dt">minClusterSize =</span> <span class="dv">40</span>,
  <span class="dt">deepSplit =</span> <span class="dv">3</span>,
  <span class="dt">pamStage =</span> <span class="ot">TRUE</span>,
  <span class="dt">maxPamDist =</span> <span class="fl">0.90</span>
)</code></pre></div>
<pre><code>##  ..cutHeight not given, setting it to 0.993  ===&gt;  99% of the (truncated) height range in dendro.
##  ..done.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">colorList_human[[<span class="st">&quot;HybridCut2&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">labels2colors</span>(humanHybridCut_<span class="dv">2</span><span class="op">$</span>labels)
MEList_human[[<span class="st">&quot;HybridCut2&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">moduleEigengenes</span>(human_genes, colorList_human<span class="op">$</span>HybridCut2) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">use_series</span>(<span class="st">&quot;eigengenes&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">orderMEs</span>()</code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="directory-setup.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
